### PR TITLE
Merge events into main

### PR DESCRIPTION
**Pull request by kolbytn**
- event system
- bot profiles
- recursive handleChat function
- sendThought function

The sendThought function doesn't have the desired effect yet. The bot mainly ignores the internal thoughts I send it. I think we do need to switch back to having a dedicated coding/acting pipeline and that may fix this and many other problems.

### CODE REVIEW

File: agent.js [hunk 1]
```diff
@@ -26,64 +29,70 @@ export class Agent {
...
+                this.history.add(username, message);
+                this.handleMessage();
             });
 
-            if (autostart)
-                this.respond('system', 'Agent process restarted. Notify the user and decide what to do.');
-            
+            if (init_message) {
+                this.history.add('system', init_message);
+                this.handleMessage();
```
Comment (MaxRobinsonTheGreat) [inline agent.js:38]: seems confusing/dangerous to add to history then call handleMessage. Why can't we do that inside the function?

**Review by MaxRobinsonTheGreat — CHANGES_REQUESTED** [review]
It seems to get confused very often when asked to do simple things. Example conversation taken from history:
```
"sumixam0: collect some dirt"
"Sure! I will collect some dirt for you.\n```\n// I am going to collect 10 dirt blocks\nawait skills.collectBlock(bot, 'dirt', 10);\n```"
"I need keep collecting and crafting to survive. I should plan what to do next and execute it."
"Code output:\nCollected 10 dirt.\nCode finished.\n"
"content": "I'm sorry, I cannot collect dirt for you."
```
In this case it successfully collected dirt, then said it couldn't do it.

I think the order is off. Shouldn't it get the code output, then respond to it ("I finished collecting stone!"), then be prompted with the "I need to keep collecting..." prompt?
Sometimes it gets stuck over and over trying to perform the same task, especially if you talk to it for a while. If you ask it to follow, then collect, then do something else you should see it getting confused/refuse to do something that it just did.

Needs more testing/tuning. I'm trying to debug it but I think you know this code better.

File: agent.js (hunk 1 above)
Comment (kolbytn) [reply agent.js:38]: We would need to only add it the first time the function is called and not during any of the recursions. I could create a wrapper function that adds to history and then calls an internalHandleMessage function.

File: utils/events.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        console.log('responding to event with code.');
+        agent.coder.queueCode(code);
+        let code_return = await agent.coder.execute();
+        console.log('code return:', code_return.message);
+        agent.history.add('system', code_return.message);
+    }
+
+    sendThought(agent, message) {
+        agent.history.add(agent.name, message);
+        agent.handleMessage();
```
Comment (MaxRobinsonTheGreat) [inline utils/events.js:42]: this isn't used right now but it should only be `agent.handleMessage(agent.name, message)`

**Review by MaxRobinsonTheGreat — APPROVED** [review]

**Pull request merged**